app/src/main/java/org/simple/clinic/bloodsugar/unitselection/BloodSugarUnitSelectionInit.kt

```diff
@@ -0,0 +1,11 @@
+package org.simple.clinic.bloodsugar.unitselection
+
+import com.spotify.mobius.First
+import com.spotify.mobius.Init
+
+class BloodSugarUnitSelectionInit : Init<BloodSugarUnitSelectionModel, BloodSugarUnitSelectionEffect> {
+  override fun init(model: BloodSugarUnitSelectionModel): First<BloodSugarUnitSelectionModel, BloodSugarUnitSelectionEffect> {
+    val effect = mutableSetOf(PreFillBloodSugarUnitSelected(model.bloodSugarUnitPreference))
```

Comment:
This is also why we need to update the model when the preference changes because if the screen gets restored _after_ the user has changed the selection, but _before_ the "Done" button is clicked, the preference will be reset back to the initial one instead of the one that the user has selected.